src/dfx/src/util/mod.rs

```diff
@@ -87,6 +88,9 @@ pub fn print_idl_blob(
             }
             if output_type == "idl" {
                 println!("{:?}", result?);
+            } else if output_type == "json" {
+                let json = convert_all(&result?, &Idl2JsonOptions::default())?;
```

Comment:
It seems that we only have one usage of `convert_all` here.
So the `idl2json_options` is always set with `Idl2JsonOptions::default()`.
Then in `convert_one`:
```
    (if idl2json_options.compact {
        serde_json::to_string
    } else {
        serde_json::to_string_pretty
    })(&json_value)
```
It always go into the second branch (`idl2json_options.compact == false`).

What about removing the `idl2json_options` argument from both `convert_one` and `convert_all`? The implementation will be simplified a lot.

Or do we want to let the user be able to set `compact` in the future? If so then it makes sense to keep the current implementation.

Reply:
Yeah, I considered this too..  I'll make the change